File: Arrays 101/941. Valid Mountain Array/valid_mountain.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        max = -1
        max_idx = -1
        for i, item in enumerate(arr):
            if arr[i] > max:
                max = arr[i]
                max_idx = i

        if max_idx == 0 or max_idx == len(arr) - 1:
            return False

        i = 1
        while i <= max_idx:
            if arr[i] <= arr[i - 1]:
                return False
            i += 1

        while i < len(arr):
            if arr[i] >= arr[i - 1]:
                return False
            i += 1
        return True

File: 1. Arrays 101/941. Valid Mountain Array/valid_mountain.cpp
class Solution
{
public:
    bool validMountainArray(vector<int> &arr)
    {
        int max = -1;
        int max_idx = -1;
        for (int i = 0; i < arr.size(); i++)
        {
            if (arr[i] > max)
            {
                max = arr[i];
                max_idx = i;
            }
        }
        if (max_idx == 0 || max_idx == arr.size() - 1)
        {
            return false;
        }
        int i = 1;
        while (i <= max_idx)
        {
            if (arr[i] <= arr[i - 1])
            {
                return false;
            }
            i++;
        }
        while (i < arr.size())
        {
            if (arr[i] >= arr[i - 1])
            {
                return false;
            }
            i++;
        }
        return true;
    }
};